### ISSUE TITLE
Network agent keeps flipping from gcm=false to gcm=true

### ISSUE THREAD

**Issue by bhinkle-work** [rancher/rancher#3528]
We upgraded from Rancher v0.51.0 to v0.56.1 and are experiencing a problem with managed networking. We have multiple environments with CentOS 7.0 hosts and are using the managed network for our application stacks.

The behavior we are observing is that, unpredictably, a network agent on one of the hosts will start rancher-net with **gcm=true** instead of **gcm=false** or vice versa (i.e. there is a mismatch on one host versus the others). This prevents managed networking communication between hosts. Generally a restart of the network agent will cause it to pick up the same setting as the rest of the environment - sometimes we have to restart it on the one-off host, sometimes we have to restart it on all the others.

I'm concerned, because:
- this is happening unpredictably
- **gcm=false** seems more common and yet  @deniseschannon believes (per #3479) that gcm=true is the ideal case for centos and ubuntu
- the network agent gets recreated every time we deploy a stack with load balancers, so it hurts us frequently
- host reboots and container deletions have not solved this problem

What else should I be looking into?

**Comment by wundram** [comment]
We did track this down to the method used to see if GCM is supported by constructing a test tunnel to 1.1.1.1.   I think this is already fixed in the RC build.   

We also have really ugly hack that seems to be helping, that we can use until the next official release.  This is pretty invasive, it checks the network agent each minute for the wrong GCM setting, and if found removes any test tunnels, and kills the rancher-net process.   The network agent will then restart the rancher-net process with the correct GCM setting.   

I wouldn't try this unless you really can't wait for the proper fix.   Running this script will set up the cron job on the host.  You need to run it as root on each rancher host.   To remove it, `rm /etc/cron.d/rancher-fix`

```bash
cat > /root/netmon <<EOF
#!/bin/bash

agent=`docker ps --filter "label=io.rancher.container.system=NetworkAgent" -q | head -1`
echo "Agent is \$agent"
procs=`docker exec \$agent ps -aux | grep rancher-net`
#echo "Procs: \$procs"
if [[ \$procs =~ gcm=false ]]; then
        echo "fixing gcm"
        procnum=`echo \$procs | awk '{print \$2;}'`
        docker exec \$agent ip xfrm state del src 1.1.1.1 dst 1.1.1.1 spi 42 proto esp
        docker exec \$agent kill -9 \$procnum
fi;
EOF

chmod +x /root/netmon

cat >  /etc/cron.d/rancher-fix <<EOF
* * * * * root /root/netmon > /dev/null 2>&1 || true
EOF
```

**Comment by ibuildthecloud** [comment]
We still have the naive check to detect GCM in place (I really wish there was a better way.  If anybody knows how please let me know).  I've just increased how hard we try.  We now loop 6 times doing remove rule/add rule.  After 6 tries if it never succeeds we say there is no GCM.  If it succeeds once we say GCM=true.  https://github.com/rancher/rancher-net/commit/5767b68391d30acd246a9799741a6b4de9155664

**Comment by deniseschannon** [comment]
@bhinkle-work @wundram This was addressed in v1.0.0-rcX releases. We currently do not support upgrades for the RCs yet, but if you still see these issues after upgrading (we should be supporting upgrades in v1.0.0-rc3 for testing), please re-open.
